src/invidious/yt_backend/extractors.cr

```diff
@@ -722,9 +722,9 @@ private module Extractors
       target.dig("primaryContents", "sectionListRenderer", "contents").as_a.each do |node|
         if new_node = node["itemSectionRenderer"]?
           raw_items << new_node["contents"].as_a
-        end
-        if node["continuationItemRenderer"]?
-          raw_items.push([node])
+        elsif node["continuationItemRenderer"]?
+          # we put the node in an array so the ContinuationItemRendererParser is able to parse it
+          raw_items << [node]
```

Comment:
That's not needed ^^ We flatten the array right after this loop ^^
(See four lines after this patch)

Reply:
I get this error when I remove the square brackets `Error: expected argument #1 to 'Array(Array(JSON::Any))#<<' to be Array(JSON::Any), not JSON::Any`

Reply:
Oh! I see the problem! you need to do the following:

```crystal
raw_items += new_node["contents"].as_a
```

Then everything will be the same type!